pkg/services/accesscontrol/acimpl/service.go

```diff
@@ -240,19 +244,16 @@ func (s *Service) getCachedUserPermissions(ctx context.Context, user identity.Re
 		return nil, err
 	}
 
-	permissions := make([]accesscontrol.Permission, 0, len(basicRolesPermissions)+len(teamsPermissions)+len(userPermissions))
-	permissions = append(permissions, basicRolesPermissions...)
-	permissions = append(permissions, teamsPermissions...)
 	permissions = append(permissions, userPermissions...)
+
 	return permissions, nil
 }
 
-func (s *Service) getCachedBasicRolesPermissions(ctx context.Context, user identity.Requester, options accesscontrol.Options) ([]accesscontrol.Permission, error) {
+func (s *Service) getCachedBasicRolesPermissions(ctx context.Context, user identity.Requester, options accesscontrol.Options, basicRolesPermissions []accesscontrol.Permission) ([]accesscontrol.Permission, error) {
```

Comment:
FYI @jalevin I renamed this from `perms` so it causes no diffs inside the method body

Reply:
@hairyhenderson I'd personally find it a bit more clear if we had permissions there :grin: 
```suggestion
func (s *Service) getCachedBasicRolesPermissions(ctx context.Context, user identity.Requester, options accesscontrol.Options, permissions []accesscontrol.Permission) ([]accesscontrol.Permission, error) {
```

Reply:
@gamab sure, but then line 261 would shadow it and it'd be confusing 😓

My goal was to reduce the diff to be as small as possible, but it's fair to rename too for clarity I guess...

Reply:
I renamed to perms in function body (short lived var) and permissions as var passed in